### DISCUSSION TITLE
Problem with adding Admin field to Search tools on Admin List view

### DISCUSSION THREAD

**Discussion by duetnet** [joomengine/Joomla-Component-Builder#936] [Q&A]
Hi there

I have an ADMIN Edit view with 31 fields of which  
- 4 fields above TABS
- 2 Fields below TABS
- 15 fields on TAB 1
- 10 fields on TAB 2
- TAB 3 and 4 are linked views

I have an ADMIN LIST view I have 5 columns where I list 12 fields (some fields are joined with "FIELD RELATIONS")
I added 5 fields (which appear on TAB1 and ABOVE TABS) to the "Filter" view option. 
The Search Tools options on the ADMIN LIST view correctly displays the fields selected.

I then tried to add fields from the 2nd TAB of the ADMIN EDIT view to the FILTER option, but they are ignored by the compiler.

I have tried to move one of these fields onto the 1st TAB of the ADMIN EDIT view, but this make no difference, it is still ignored and does not appear in the "Search Tools" options. I also tried to move it to BELOW TABS and LEFT of TABS, makes no difference.

When I remove one of the original fields from the FILTER list, it will disappear from the "Search tools" option.
When I add it back, it appears again.

However, I cannot get ANY of these fields from the 2nd Tab to appear in the "Search Tools", even if I ONLY select them and NONE of the original fields from the 1st TAB.

I have monitored the source of the relevant view.html.php file with each change but these fields just REFUSE to be added to the "SEARCH tools" list.
I am a bit stumped. Anyone have any suggestions?

(Environment: Win 64, PHP 8.0.8/5.5.5-10.4.20-MariaDB / Joomla 3.10.11 / JCB 3.1.9)

**Comment by Llewellynvdm** [comment]
The probability that these are not search type fields... since some fields are not easy search fields, and need extra help to work. So lets start with what field types these fields are, and what storage method is used. Then what kind of database type fields are these, as only some of them work for search.

Should we find the this is a bug... then you need to open an issue on https://git.vdm.dev/joomla/Component-Builder/issues

But let me say that I have never had these kind of issues, as the tabs play no roll in the search tool setup what so ever. But field types and data types do... look here, here the filter values are set: (https://git.vdm.dev/joomla/Component-Builder/src/branch/staging/admin/helpers/compiler/c_Fields.php#L5271) and here you will see when they are not allowed: (https://git.vdm.dev/joomla/Component-Builder/src/branch/staging/admin/helpers/compiler/c_Fields.php#L5239) as you can see their tab placement has nothing todo with that.

**Comment by duetnet** [comment]
Hi Llewellyn

 

Ok, I think I understand what is happening…

 

I debugged the compiling process and I believe I now may understand it better….

 

The following 2 pieces of code in c_fields.php is where I eventually realised what is happening…

 



 

And

 



 

 

From the above I have deduced that if the field I wish to filter (i.e. $field[‘filter’] is set and is >=1) will still be ignored if it is not:

 

*	Listed directly on the ADMIN LIST view ($listSwitch) or
*	Listed indirectly via the “FIELD RELATIONS” definition($listJoin)  of the ADMIN LIST view or
*	is one of the field types checkbox/es, repeatable or subform 

 

What I did establish was that I can get a field included in the in the create the following work around…

 

If I do not what the field listed directly or indirectly on the list view, but I still need a filter for it, then I can include it in the “FIELD RELATIONS” definition of another field in the “JOIN FIELD” definition and then using the “Area”=”View”, Join Type=”Custom>>Code” and then merely REMOVE the field from the “Set” definition.

 

Although not critical to my current requirements, I find an issue does arise with where I try and use a list field (which I have configured as per above and so is not listed on the main admin list view) which has LANGUAGE TEXT definitions, because the compiler will generate an array in the “MODELS\ADMIN_VIEWS.PHP” code changing the text (language strings interpretation) in the array from something like “COM_COMPONENTNAME_JANUARY” to “COM_ADMINVIEWNAME_COMPONENTNAME_JANUARY”. This results in warning message on the list view “UNDEFINED ARRAY KEY”.

 

Anyways, thanks for your feedback, I do now have a better understanding of what is required for the FILTERS.

 

Regards

 

Franz
